tests/js/common/shell/shell-index-ensure.js

```diff
@@ -107,6 +112,76 @@ function ensureIndexSuite() {
       assertEqual(collection.name() + "/" + id, res.id);
     },
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test: names
+////////////////////////////////////////////////////////////////////////////////
+
+    testEnsureNamePrimary : function () {
+      var res = collection.getIndexes()[0];
+
+      assertEqual("primary", res.type);
+      assertEqual("primary", res.name);
+    },
+
+    testEnsureNameEdge : function () {
+      var res = edgeCollection.getIndexes()[0];
+
+      assertEqual("primary", res.type);
+      assertEqual("primary", res.name);
+
+      res = edgeCollection.getIndexes()[1];
+
+      assertEqual("edge", res.type);
+      assertEqual("edge", res.name);
+    },
+
+    testEnsureName1 : function () {
+      var name = "byValue";
+      var idx = collection.ensureIndex({ type: "skiplist", fields: [ "b", "d" ], name: name });
+      assertEqual("skiplist", idx.type);
+      assertFalse(idx.unique);
+      assertEqual([ "b", "d" ], idx.fields);
+      assertEqual(name, idx.name);
+
+      var res = collection.getIndexes()[collection.getIndexes().length - 1];
+
+      assertEqual("skiplist", res.type);
+      assertFalse(res.unique);
+      assertEqual([ "b", "d" ], res.fields);
+      assertEqual(name, idx.name);
+    },
+
+    testEnsureName2 : function () {
+      var name = "byValue";
+      var idx = collection.ensureIndex({ type: "skiplist", fields: [ "b", "d" ], name: name });
+      assertEqual("skiplist", idx.type);
+      assertFalse(idx.unique);
+      assertEqual([ "b", "d" ], idx.fields);
+      assertEqual(name, idx.name);
+
+      // expect duplicate name to fail and return old index with same name
+      idx = collection.ensureIndex({ type: "hash", fields: [ "a", "c" ], name: name });
+      assertEqual("skiplist", idx.type);
+      assertFalse(idx.unique);
+      assertEqual([ "b", "d" ], idx.fields);
```

Comment:
Is this actually the expected behavior?
If the index is created with a different definition than an existing index but using a name of an already existing index, I think the request should fail. This is of course debatable, but returning an index on different fields I find somewhat unexpected.